crates/core/src/host/module_host.rs

```diff
@@ -760,11 +755,33 @@ impl ModuleHost {
         let db = self.inner.replica_ctx().relational_db.clone();
         // scheduled reducer name not fetched yet, anyway this is only for logging purpose
         const REDUCER: &str = "scheduled_reducer";
+        let caller_identity = self.info.identity;
+        let caller_address = self.info.address;
+        let module = self.info.clone();
         self.call(REDUCER, move |inst: &mut dyn ModuleInstance| {
-            let mut tx = db.begin_mut_tx(IsolationLevel::Serializable);
+            let mut tx = db.begin_mut_tx(IsolationLevel::Serializable, Workload::Internal);
```

Comment:
Could we pass `Workload::Reducer`, but have the reducer context be `None` until we fetch the name?

Reply:
Will end up as before this suggestion.